spec/statement.dd

```diff
@@ -511,32 +511,38 @@ $(H3 $(LNAME2 foreach_over_arrays, Foreach over Arrays))
 $(P
         If the aggregate is a static or dynamic array, there
         can be one or two variables declared. If one, then the variable
-        is said to be the $(I value) set to the elements of the array,
-        one by one. The type of the
-        variable must match the type of the array contents, except for the
-        special cases outlined below.
+        is said to be the $(I value), which is set successively to each
+        element of the array. The type of the variable, if specified,
+        must be compatible with the array element type (except for the
+        special handling of character elements outlined below).
+        The *value* variable can modify array elements when
+        $(RELATIVE_LINK2 foreach_ref_parameters, declared with `ref`).
+)
+$(P
         If there are
         two variables declared, the first is said to be the $(I index)
-        and the second is said to be the $(I value). The $(I index)
-        must be of type `size_t` for dynamic arrays. Static arrays may use
-        any integral type that spans the length of the array.
-        $(I index) cannot be `ref`.
-        It is set to be the index of the array element.
+        and the second is said to be the $(I value) as above.
+        $(I index) cannot be declared with `ref`.
+        It is set to the index of the array element on each iteration.
+        The index type can be inferred:
 )
 $(SPEC_RUNNABLE_EXAMPLE_RUN
 --------------
 char[] a = ['h', 'i'];
 
-foreach (size_t i, char c; a)
+foreach (i, char c; a)
 {
     writefln("a[%d] = '%c'", i, c);
 }
 --------------
 )
+        $(P The $(I index) type must be `size_t` for a dynamic array.
```

Comment:
This is not true, it must be an integral type that a `size_t` converts to. It can also be a signed `long` for example.

Reply:
Thanks, fixed.